sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -54,8 +54,14 @@
         result = await serialize_property(v, deps, transform)
         # We treat properties that serialize to None as if they don't exist.
         if result is not None:
+            # While serializing to a pb struct, we must "translate" all key names to be what the engine is going to
+            # expect. Resources provide the "transform" function for doing this.
+            translated_name = k
+            if transform is not None:
```

Comment:
`transform` is an awfully generic name. I know it's a largish refactoring, but could we rename it to something clearer? Is there ever a scenario where we would expect the transform to be something besides `res.translate_input_property`?

Reply:
> Is there ever a scenario where we would expect the transform to be something besides res.translate_input_property?

nvm on this point, I see that invoke will need to provide its own implementation here.

Reply:
> Is there ever a scenario where we would expect the transform to be something besides res.translate_input_property?

Nope, other than unit testing. I'm testing this code using langhost-level integration tests, though, not RPC-specific unit tests.

Reply:
I can do that.